pywren/pywren_ibm_cloud/wren.py

```diff
@@ -253,237 +238,250 @@ def wait(self, throw_except=True, verbose=True, return_when=ALL_COMPLETED,
-                
+
+                still_not_done_futures = [f for f in futures if ((f.callgroup_id, f.call_id) in not_done_call_ids)]
+
                 if verbose and still_not_done_futures:
                     pool.map(fetch_future_results, still_not_done_futures)
                 elif still_not_done_futures:
-                    futures = pool.map(fetch_future_results, still_not_done_futures)            
+                    futures = pool.map(fetch_future_results, still_not_done_futures)
                     for f in futures:
                         if f.done:
                             pbar.update(1)
                     pbar.refresh()
-    
+
                 callids_done_in_callset.update([(f.callgroup_id, f.call_id) for f in still_not_done_futures if f.done])
-    
+
             if not verbose:
                 pbar.close()
                 print()
             pool.close()
-            self._state = ExecutorState.success
-        
+            self._state = ExecutorState.finished
+
         except (TimeoutError, IndexError):
             if not verbose:
                 if pbar:
                     pbar.close()
-                    print() 
-            not_dones_activation_ids = set([f.activation_id for f in self.futures if not f.done])
-            msg='Executor ID {} Raised timeout of {} seconds getting results \nActivations not done: {}'.format(self.executor_id, timeout, not_dones_activation_ids)
+                    print()
+            not_dones_activation_ids = set([f.activation_id for f in futures if not f.done])
+            msg = ('Executor ID {} Raised timeout of {} seconds getting results '
+                   '\nActivations not done: {}'.format(self.executor_id, timeout, not_dones_activation_ids))
             logger.info(msg)
             if(logger.getEffectiveLevel() == logging.WARNING):
                 print(msg)
             self._state = ExecutorState.error
-        
+
         except KeyboardInterrupt:
             if not verbose:
                 if pbar:
                     pbar.close()
                     print()
-            not_dones_activation_ids = [f.activation_id for f in self.futures if not f.done]
-            msg='Executor ID {} Cancelled  \nActivations not done: {}'.format(self.executor_id, not_dones_activation_ids)
+            not_dones_activation_ids = [f.activation_id for f in futures if not f.done]
+            msg = 'Executor ID {} Cancelled  \nActivations not done: {}'.format(self.executor_id, not_dones_activation_ids)
             logger.info(msg)
             if(logger.getEffectiveLevel() == logging.WARNING):
                 print(msg)
             exit()
-        
+
         finally:
             if not verbose:
                 if pbar:
                     pbar.close()
             signal.alarm(0)
-            if (self.data_cleaner):
+            if self.data_cleaner and not self._openwhisk:
```

Comment:
@JosepSampe we do use this over openwhisk. We need clean there also

Reply:
@gilv When IBM-PyWren is used inside a CF, it is used the same **executor_id** generated in the client side. It allows to better track all the functions used in an experiment launched by a client, even if it is a function which spawns new functions. Thus, the responsible to automatically delete all the data when the whole experiment finishes is the client. 
There may be cases that execute the cleaner within a function could delete important data before finish the experiment, which could lead to the client's side getting stuck.

Reply:
@JosepSampe my question is why 
if self.data_cleaner and not self._openwhisk: and not only if self.data_cleaner?
Or self._openwhisk relevant both to CF and vanilla OpenWhisk?

Reply:
@gilv Maybe the name of the variable is a bit confusing. This variable contains False when the client runs PyWren in his machine, and True when PyWren is executed within a function in CF cluster. I can change the variable to `self.cf_cluster` or something like that.

What I pretend with `if self.data_cleaner and not self._openwhisk:` is to run the cleaner only in the client machine, and not when PyWren is executed from within a function. This because the client will also delete this data at the end of the execution.

Reply:
@JosepSampe got it...the variable name confused me. You are right, it's better change the name to self.cf_cluster

Reply:
@gilv I created a new commit with the changes 7f02edf